src/lib/InstantSearch.js

```diff
@@ -121,19 +121,27 @@ Usage: instantsearch({
     );
 
     if (this._searchFunction) {
-      this.helper = Object.create(helper);
-      this.helper.search = () => {
-        helper.setState(this.helper.state);
-        this._searchFunction(helper);
+      this._searchToAlgolia = helper.search.bind(helper);
```

Comment:
Now that I read it, I feel like we could name this "this._mainHelperSearch" rather than searchToAlgolia.